File: 프로그래머스 풀이/lv2_위장.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 0;
    string kind[31];
    int num[31] = {0,};
    bool ishave = false;
    int i, j, idx=0, size = clothes.size();
    
    for (i = 0; i < size; i++) {
        ishave = false;
        for ( j=0; j < idx; j++) {
            if ( kind[j] == clothes[i][1] ){
                num[j] += 1;
                ishave = true;
                break;
            }
        }
        if ( !ishave ){
            kind[idx] = clothes[i][1];
            num[idx++] = 1;
        }
    }
    int mul = 1;
    for (i = 0; i < idx; i++)
        if ( num[i] != 0)   mul *= (num[i]+1);
    
    answer = mul - 1;
    return answer;
}

File: 프로그래머스 풀이/lv2_위장.java
import java.util.ArrayList;
class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        ArrayList<String> list_str = new ArrayList<String>();
        ArrayList<Integer> list_num = new ArrayList<Integer>();
        
        int i =0;
        int index = -1;
        while ( i < clothes.length) {
            index = list_str.indexOf(clothes[i][1]);
            if ( index == -1 ) {
                list_str.add(clothes[i][1]);
                list_num.add(1);
            }
            else {             
                list_num.set(index, list_num.get(index) + 1);
            }
            i++;
        }
        //  ( nC1 + nC0 ) *... 모든 경우 곱하기
        for(int k = 0; k < list_num.size(); k++) {
            answer  *= ( list_num.get(k) + 1 );     
        }
        // 안 입을 경우 1빼기
        answer -= 1;
        return answer;
    }    
}